src/tcia_utils/nbia.py

```diff
@@ -910,6 +928,153 @@ def getDoiMetadata(doi, output, api_url = "", format = ""):
+    manufacturers: list[str] -- Imaging device manufacturers, e.g. SIEMENS
+    bodyParts: list[str]     -- Body parts of interest, e.g. CHEST, ABDOMEN
+    fromDate: str            -- First cutoff date, in DD/MM/YYYY format. Defaults to 1 January 1900
+    toDate: str              -- Second cutoff date, in DD/MM/YYYY format. Defaults to today's date
+    patients: list[str]      -- Patients to include in the output
+
+    Example call: getSimpleSearchCriteriaValues(collections=["TCGA-UCEC", "4D-Lung"], modalities=["CT"])
+    """
+
+    endpoint = "getSimpleSearchCriteriaValues"
+    criteriaTypeIndex = 0
+    options = {}
+
+    getCriteria = lambda: "".join(['criteriaType', str(criteriaTypeIndex)])
+    getValue = lambda: "".join(['value', str(criteriaTypeIndex)])
+    getCriteriaAndValue = lambda: (getCriteria(), getValue())
+    def setOptionValue(criteriaType, param):
+        criteria, value = getCriteriaAndValue()
+        options[criteria] = str(criteriaType.value)
+        options[value] = param
+
+    if fromDate or toDate:
+        from_date, to_date, bad_dates = None, None, []
+        if toDate and not fromDate:
+            print("No fromDate specified, using 1 January 1900")
+            fromDate = "01/01/1900"
+        if fromDate and not toDate:
+            print("No toDate specified, using today's date")
+            toDate = datetime.now().strftime("%d/%m/%Y")
+        try:
+            from_date = datetime.strptime(fromDate, "%d/%m/%Y")
+        except ValueError:
+            bad_dates.append("fromDate")
+        try:
+            to_date = datetime.strptime(toDate, "%d/%m/%Y")
+        except ValueError:
+            bad_dates.append("toDate")
+        if bad_dates:
+            print(f'Malformed date parameter(s) {bad_dates}; use d/m/Y format e.g. 31/12/1999')
+            raise StopExecution
+
+    if collections:
+        for collection in collections:
+            setOptionValue(Criteria.Collection, collection)
+            criteriaTypeIndex += 1
+    if species:
+        for val in species:
+            setOptionValue(Criteria.Species, NPEXSpecies[val])
+            criteriaTypeIndex += 1
+    if modalities:
+        for modality in modalities:
+            setOptionValue(Criteria.ImageModality, modality)
+            criteriaTypeIndex += 1
+    if bodyParts:
+        for bodyPart in bodyParts:
+            setOptionValue(Criteria.BodyPart, bodyPart)
+            criteriaTypeIndex += 1
+    if manufacturers:
+        for manufacturer in manufacturers:
+            setOptionValue(Criteria.Manufacturer, manufacturer)
+            criteriaTypeIndex += 1
+    if patients:
+        for patient in patients:
+            setOptionValue(Criteria.Patient, patient)
+            criteriaTypeIndex += 1
+    if minStudies:
+        setOptionValue(Criteria.NumStudies, minStudies)
+        criteriaTypeIndex += 1
+    if modalityAnded:
```

Comment:
After further testing, this may not have anything to do with the modalityAnded thing.  Even running getSimpleSearchCriteriaValues() with no parameters I get significantly lower numbers in the results than I see in the GUI with no search filters selected.  You can reproduce by looking at the numbers next to each item in the Simple Search left side menu at https://nbia.cancerimagingarchive.net/nbia-search/.

Reply:
@zjp I'm not sure if this discrepancy in search results compared to the NBIA GUI was ever something you were able to reproduce.  Also, I think it needs some updates to use the logging approach instead of print messages.  Can you take a look and let me know when you think you've got those things sorted out?

Reply:
I updated the code to use the logging approach, but I'm having trouble querying that endpoint -- not the whole advanced API, I don't think, just the one; it times out -- so I'm not able to look at it for now.

Reply:
Perhaps it would be useful to do away with modalityAnded for now and follow up on a later ticket. If I do

`nbia.getSimpleSearchCriteriaValues(collections=["4D-Lung"])`

I do get the correct values: 1 collection, 20 subjects, 1 species, 2 modalities, 1 anatomical site, 1 manufacturer.

Reply:
4D-Lung was one of the ones that looked okay to me before.  I tried with no parameters so it would just pull back everything and let me quickly scan the patient counts for the various collections/modalities/etc.  The numbers in the GUI were different on a decent percentage, but not all.

Reply:
Do you have any examples? When I try to query with no parameters, I get a gateway timeout -- understandable, since that's a lot of data to request. But it means I can't just go down the whole list all at once.

Reply:
Huh. I honestly don't see how it could be a client-side issue, but I can see that the simple search GUI uses a different endpoint to display results (for paging), and that endpoint's code is substantially shorter than this one's. 

See what the GUI uses
https://github.com/CBIIT/NBIA-TCIA/blob/master/software/nbia-api/src/gov/nih/nci/nbia/restAPI/GetSimpleSearchWithModalityAndBodyPartPaged.java

vs what I query here
https://github.com/CBIIT/NBIA-TCIA/blob/master/software/nbia-api/src/gov/nih/nci/nbia/restAPI/GetSimpleSearchCriteriaValues.java

Reply:
Based on https://wiki.cancerimagingarchive.net/display/Public/NBIA+Advanced+REST+API+Guide#NBIAAdvancedRESTAPIGuide-ExtendedSimpleSearchwithModalityandBodyPartPagedAPI, I think https://github.com/CBIIT/NBIA-TCIA/blob/master/software/nbia-api/src/gov/nih/nci/nbia/restAPI/GetSimpleSearchWithModalityAndBodyPartPaged.java is what's retrieving the information that appears in the search results on the right side of that screenshot.  I am not sure what API is pulling the numbers that appear next to the search criteria on the left side menu, but I was thinking it might be [GetSimpleSearchCriteriaValues.java](https://github.com/CBIIT/NBIA-TCIA/blob/master/software/nbia-api/src/gov/nih/nci/nbia/restAPI/GetSimpleSearchCriteriaValues.java).   I've emailed the lead developer of the GUI to weigh in.

Reply:
Actually, after you pointed me to looking for clues in the Network tab in your other PR I gave that a try for investigating this one.  I don't see [GetSimpleSearchCriteriaValues.java](https://github.com/CBIIT/NBIA-TCIA/blob/master/software/nbia-api/src/gov/nih/nci/nbia/restAPI/GetSimpleSearchCriteriaValues.java) in the list but I do see the separate calls for modality w/ counts, body part w/ counts, etc (https://nbia.cancerimagingarchive.net/nbia-api/services/getBodyPartValuesAndCounts) so perhaps he's just running several separate calls to get the left menu numbers.

Reply:
He confirmed he doesn't use [GetSimpleSearchCriteriaValues.java](https://github.com/CBIIT/NBIA-TCIA/blob/master/software/nbia-api/src/gov/nih/nci/nbia/restAPI/GetSimpleSearchCriteriaValues.java) anymore.  

I will create a bug report to have someone investigate both the gateway timeout and the inaccurate results that it returns.

Reply:
If the intention was to deprecate that method then I can pivot to querying the endpoint that's in use (getSimpleSearchWithModalityAndBodyPartPaged). It returns all of the same information that I wanted, plus some data that can be ignored.